app/Services/Manager.php

```diff
@@ -418,4 +418,22 @@ public function appendReportbackItemToMessage($reportback_id, $reportback_item_i
         // Find the matching reportback item, and return the item.
         return $reportback_items->get($reportback_item_id);
     }
+
+    /**
+     * Attach a user object to a waiting room or a competition.
+     *
+     * @param string $type
+     * @param string $id
+     * @param object $user
+     * @return bool
+     */
+    public function addUserToRoom($type, $id, $user)
```

Comment:
Nice! Maybe call it `addUserToModel` and switch `$type` to `$model`. "room" is going to easily be confused with just the waiting room.